### ISSUE TITLE
Thread reply with concierge gets automatically pinned even if the original chat is not pinned

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22071]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open a new account
2. Unpin the concierge
3. Hover over the first message from concierge and reply in thread
4. Go back and open the thread again by clicking on the "1 Reply" button

## Expected Result:
The thread should not be automatically pinned

## Actual Result:
Thread replies from concierge gets pinned

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.35-5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/8added2d-710f-4a96-aa48-20fbb7dd09b9

https://github.com/Expensify/App/assets/93399543/6d46f532-b0e4-4e12-8ed1-d62e32594abe

**Expensify/Expensify Issue URL:**
**Issue reported by:** React JS development and Quality Assurance (GH @Nathan-Mulugeta)
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688209194465719

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01edf19e94d20b64a5</li>
        <li>Upwork Job ID: 1676256084803575808</li>
        <li>Last Price Increase: 2023-07-04</li>
    </ul>
</details>

**Comment by StevenKKC** [comment]
This seems BE issue.
After reply in thread, BE first responds `report` with `isPinned: false`, and if reopen the thread again, BE responds `report` with `isPinned: true`.

**Comment by trjExpensify** [comment]
Yup, I can reproduce this one. I don't think a thread in the Concierge DM should be automatically pinned in the same way that a thread in any other chat (pinned or not) doesn't assume the parent chat's pinned status. 

<img width="1511" alt="image" src="https://github.com/Expensify/App/assets/16232057/939959c3-3881-4fff-97eb-362bcb8ad56e">

**Comment by trjExpensify** [comment]
Based on [this comment](https://github.com/Expensify/App/issues/22071#issuecomment-1618494093), adding `internal` as a backend issue.

**Comment by Julesssss** [comment]
It seems unlikely that this is only occurring on web, did we not test the other platforms?

**Comment by trjExpensify** [comment]
Ah yes, I forgot to remove the web specific piece from the OP. It's reproducible on on all.

**Comment by Julesssss** [comment]
I ran out of time to triage this today, it's top of the list for tomorrow

**Comment by Julesssss** [comment]
I had another insane day. Will get to this soon for triage

**Comment by Julesssss** [comment]
Finally done with deploy dyte and found time. Agree it's likely backend and will investigate further when I have a moment.

**Comment by Julesssss** [comment]
Investigating this today

**Comment by Julesssss** [comment]
I fixed this with a[ backend PR](https://github.com/Expensify/Web-Expensify/pull/38180), which can't be tested by C+ team.

**Comment by Julesssss** [comment]
Has been deployed to prod, closing

**Comment by Julesssss** [comment]
Wait, we need to pay reporting bonus to @Nathan-Mulugeta in two days time.

**Comment by Nathan-Mulugeta** [comment]
Hey @Julesssss payment has not been processed yet.

**Comment by Nathan-Mulugeta** [comment]
Bump to remind about payment @Julesssss

**Comment by trjExpensify** [comment]
Ah woof, the header didn't update and I missed it on the 27th. Apols! @Nathan-Mulugeta I've sent you an offer on this job [here](https://www.upwork.com/jobs/~01edf19e94d20b64a5).

**Comment by Nathan-Mulugeta** [comment]
Just accepted offer @trjExpensify
